theano/sparse/basic.py

```diff
@@ -655,31 +626,22 @@ def __init__(self, format, kmap=None):
         if format not in ('csr', 'csc'):
             raise ValueError("format must be one of: 'csr', 'csc'", format)
         self.format = format
-
-        # for efficiency, if remap does nothing, then do not apply it
-        if kmap is not None and all(kmap == numpy.arange(numpy.size(kmap))):
-            kmap = None
-
+        if kmap is not None:
+            raise Exception("Do not use kmap, it is removed")
         self.kmap = kmap
+        # should view the other inputs too, but viewing multiple
+        # inputs is not currently supported by the destroyhandler
+        self.view_map = {0: [0]}
 
-        if not isinstance(self.kmap, numpy.ndarray):
-            # should view the other inputs too, but viewing multiple
-            # inputs is not currently supported by the destroyhandler
-            self.view_map = {0: [0]}
-
-        self._hashval = (hash(type(self)) ^ hash(self.format) ^
-                         _kmap_hash(self.kmap))
+        self._hashval = (hash(type(self)) ^ hash(self.format))
```

Comment:
It is good. Can you go one step futur and add a \__props__ = ('format',) to this op. Then you can clean it up by removing \_hashval, \__eq__, \__hash__ and \__str__.